src/error.rs

```diff
@@ -13,16 +13,32 @@ pub enum AmbitError {
     //       Future changes may result in a Vec<ParseError> being returned.
     //       This should be taken care of.
     Parse(config::ParseError),
+    File { path: String, error: io::Error },
     Other(String),
 }
 
-impl Error for AmbitError {}
+impl Error for AmbitError {
+    fn source(&self) -> Option<&(dyn Error + 'static)> {
+        match self {
+            AmbitError::File { error, .. } => Some(error),
+            _ => None,
+        }
+    }
+}
 
 impl Display for AmbitError {
     fn fmt(&self, f: &mut Formatter) -> fmt::Result {
-        match *self {
+        match &*self {
             AmbitError::Io(ref e) => e.fmt(f),
             AmbitError::Parse(ref e) => e.fmt(f),
+            AmbitError::File { path, .. } => f.write_str(
+                format!(
+                    "Failed to read `{}`\n\nCaused by:\n  {}",
+                    path,
+                    self.source().unwrap()
```

Comment:
I think it would be better to do this for _all_ error variants (after all, they may have sources too.) This can be achieved by simply putting another `match` statement after this one.